app/uk/gov/hmrc/cataloguefrontend/service/DefaultBranchesService.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.service
+
+import uk.gov.hmrc.cataloguefrontend.connector.RepositoryDisplayDetails
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject._
+import scala.concurrent.ExecutionContext
+import scala.language.postfixOps
+
+@Singleton
+class DefaultBranchesService @Inject()()(implicit val ec: ExecutionContext){
+
+  def allTeams(repos: Seq[RepositoryDisplayDetails]): Seq[String] = {
+    val repositories = repos.map(t => t.teamNames)
+    repositories.flatten.distinct.sorted
+  }
+
+  def updatedDefaultBranchCount(
```

Comment:
Can you not just use `filterRepositories(..).count(_.defaultCount != "master"` to avoid duplicating all the logic?